Gems/Atom/Tools/AtomToolsFramework/Code/Source/EntityPreviewViewport/EntityPreviewViewportSettingsSystem.cpp

```diff
@@ -272,6 +272,54 @@ namespace AtomToolsFramework
         return result.IsSuccess() ? result.GetValue() : AZ::Data::AssetId();
     }
 
+    bool EntityPreviewViewportSettingsSystem::LoadRenderPipeline(const AZStd::string& path)
+    {
+        const auto& pathWithAlias = GetPathWithAlias(path);
+        const auto& pathWithoutAlias = GetPathWithoutAlias(path);
+        auto cacheItr = m_renderPipelineDescriptorCache.find(pathWithAlias);
+        if (cacheItr != m_renderPipelineDescriptorCache.end())
+        {
+            SetSettingsValue("/O3DE/AtomToolsFramework/EntityPreviewViewportSettings/RenderPipelinePath", pathWithAlias);
```

Comment:
Nit: the setreg key is repeated a few times here, might be useful to have as a constant to avoid typos/errors later

Reply:
There is a lot of duplication throughout this file, I was on the fence about how much I want to improve it right now. @gadams3 what do you think?

Reply:
You're welcome to do it if you want to. Otherwise I can do it later. This class was originally created specifically for material editor lighting and model presets prior to the document system being generalized. I would like to move lighting presets to their own tool or move a lot of this class to use the document system instead. The viewport settings panel would be replaced by the document inspector. Model presets would go away and we would just reference model assets. The viewport would just listen for asset changes or settings changes. Most of this class would disappear and the rest could live in the settings registry. All of this is in line with several GHI bugs and feature requests.

Reply:
I'll just leave it then, since you have some plans to move all this around anyway.